plugins/modules/proxmox_cluster_ha_groups.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: proxmox_cluster_ha_groups
+
+short_description: Management of HA groups in roxmox VE Cluster
+
+version_added: "1.0.2"
+
+description: 
+  - Configure HA groups via /cluster/ha/groups
+
+options:
+    state:
+        description: create or delete
+        required: true
+        choices: ['present', 'absent']
+        type: str
+    name:
+        description: group name
+        required: true
+        type: str
+    comment:
+        description: Description
+        required: false
+        type: str
+    nodes:
+        description: List of cluster node members, where a priority can be given to each node. A resource bound to a group will run on the available nodes with the highest priority. If there are more nodes in the highest priority class, the services will get distributed to those nodes. The priorities have a relative meaning only. The higher the number, the higher the priority.
+        required: false
+        type: bool
+    nofailback:
+        description: The CRM tries to run services on the node with the highest priority. If a node with higher priority comes online, the CRM migrates the service to that node. Enabling nofailback prevents that behavior.
+        required: false
+        type: bool
+    restricted:
+        description: Resources bound to restricted groups may only run on nodes defined by the group. The resource will be placed in the stopped state if no group node member is online. Resources on unrestricted groups may run on any cluster node if all group members are offline, but they will migrate back as soon as a group member comes online. One can implement a 'preferred node' behavior using an unrestricted group with only one member.
+        required: False
+        type: bool
+
+author:
+    - Markus Kötter (@commonism)
+'''
+
+EXAMPLES = r'''
+- name: Create HA group
+  community.proxmox.proxmox_cluster_ha_groups:
+    api_host: "{{ ansible_host }}"
+    api_password: "{{ proxmox_root_pw | default(lookup('ansible.builtin.env', 'PROXMOX_PASSWORD', default='')) }}"
+    api_user: root@pam
+
+    state: "present"
+    name: ha0
+    comment: yes
+    nodes: node0:0,node1:1
+    nofailback: true
+    restricted: false
+
+- name: Delete HA group
+  community.proxmox.proxmox_cluster_ha_groups:
+    api_host: "{{ ansible_host }}"
+    api_password: "{{ proxmox_root_pw | default(lookup('ansible.builtin.env', 'PROXMOX_PASSWORD', default='')) }}"
+    api_user: root@pam
+
+    state: "absent"
+    name: ha0
+'''
+
+RETURN = r'''
+# These are examples of possible return values, and in general should use other names for return values.
+old_groups:
+    description: The original name param that was passed in.
+    type: list
+    returned: always
+new_groups:
+    description: The output message that the test module generates.
+    type: list
+    returned: when changed
```

Comment:
The `RETURN` section needs to be filled out with real data. If there is no return value, use `RETURN = r'''#'''` or something like that.